packages/app/app/containers/SoundContainer/sleeptimer.js

```diff
@@ -0,0 +1,32 @@
+let props;
+let sleeptimerSleepDate;
+let sleeptimerTimeToSleep=0;
+
+export function putToSleep (callProps) {
+  props = callProps;
+  const PERMITTEDDIFFERENCE= 5; 
+
+  if (sleeptimerTimeToSleep !== props.settings.sleeptime){                      // sleep time setting was changed
+    sleeptimerSleepDate = 'undefined';
+  }
+
+  if (sleeptimerSleepDate === 'undefined'){                                    // set sleeptimerSleepDate to pause time
+    sleeptimerTimeToSleep = props.settings.sleeptime;    
+    sleeptimerSleepDate = new Date(new Date().getTime() + sleeptimerTimeToSleep * 60 * 1000);     
+  } else {                                                     
+    if (new Date(((new Date().getTime()) - sleeptimerTimeToSleep) -sleeptimerSleepDate) >= PERMITTEDDIFFERENCE*60*1000){ // this will insure that sleeptimer was not turned off then back on at a later date and instantly pause the song 
+      // (it will make sure it is turned off within PERMITTEDDIFFERENCE minutes of sleepdate being surpassed)
+      sleeptimerSleepDate = 'undefined';
+      return;
+    }
+        
+    // sleeptimerSleepDate exists and is accurate so grab currentdate and compare 
+    const currentDate = new Date(new Date().getTime());
+    if (sleeptimerSleepDate < currentDate){                                  // if sleeptimerSleepDate passed then pause song and set sleeptimerSleepDate to undefine to reset sleeptimer
+      sleeptimerSleepDate = 'undefined';
+      props.settings.sleeptimerSwitch = false;
```

Comment:
Props should be immutable. I know this is a local reference, but this state should be kept higher in the tree instead of just assigning it to a module level variable